src/containers/CookieConsentContainer/useCookieConsentContainer.ts

```diff
@@ -63,9 +45,8 @@ export const useCookieConsentContainer = (): CookieConsentProps | undefined => {
       });
       return;
     }
-    setCookieConsent(undefined);
-    startGoogleAnalytics({ isAnonymous: !cookies.hasConsent });
-  }, [acceptCookieConsent, cookies.hasConsent, declineCookieConsent, startGoogleAnalytics, t]);
+    setCookieConsentProps(undefined);
```

Comment:
maybe you can use useMemo to calculate cookieConsentProps based on cookies values instead of using useState and useEffect?